server/internal/client/client.go

```diff
@@ -110,7 +110,7 @@ import (
 // by the user, and once by this monitor goroutine.
 
 const (
-	maxOutstandingRequests = 1024
+	maxOutstandingRequests = 16384
```

Comment:
In addition to increasing the default, it's probably a good idea to make this configurable at runtime,